src/AddressSpace.cc

```diff
@@ -1040,10 +1040,16 @@ void AddressSpace::fix_stack_segment_start(const Mapping& mapping,
 }
 
 Mapping AddressSpace::vdso() const {
+  return vdso_pair().first;
+}
+
+MappingResourcePair AddressSpace::vdso_pair() const{
   assert(!vdso_start_addr.is_null());
-  return mapping_of(vdso_start_addr).first;
+  auto mapping = mapping_of(vdso_start_addr);
+  return mapping;
 }
```

Comment:
Also I think we need a test here. It's a little bit tricky to write; you probably should modify test_lib.c's `constructor` function to unmap the VDSO, by reading /proc/self/maps, finding the [vdso] line, and unmapping that page range.